app/tests/evaluation_tests/test_forms.py

```diff
@@ -222,12 +223,61 @@ def test_algorithm_with_permission(self):
         form = SubmissionForm(
             user=user,
             phase=p,
-            data={"algorithm_image": ai.pk, "creator": user, "phase": p},
+            data={"algorithm": alg, "creator": user, "phase": p},
         )
 
         assert form.errors == {}
-        assert "algorithm_image" not in form.errors
+        assert "algorithm" not in form.errors
+        assert form.is_valid()
+
+    def test_algorithm_image_and_model_set(self):
+        user = UserFactory()
+        alg = AlgorithmFactory()
+        alg.add_editor(user=user)
+        ci1 = ComponentInterfaceFactory()
+        ci2 = ComponentInterfaceFactory()
+        alg.inputs.set([ci1])
+        alg.outputs.set([ci2])
+        archive = ArchiveFactory()
+        p = PhaseFactory(
+            submission_kind=SubmissionKindChoices.ALGORITHM,
+            submissions_limit_per_user_per_period=10,
+            archive=archive,
+        )
+        p.algorithm_inputs.set([ci1])
+        p.algorithm_outputs.set([ci2])
+        civ = ComponentInterfaceValueFactory(interface=ci1)
+        i = ArchiveItemFactory(archive=p.archive)
+        i.values.add(civ)
+
+        InvoiceFactory(
+            challenge=p.challenge,
+            compute_costs_euros=10,
+            payment_status=PaymentStatusChoices.COMPLIMENTARY,
+        )
+
+        # Fetch from the db to get the cost annotations
+        # Maybe this is solved with GeneratedField (Django 5)?
+        p = Phase.objects.get(pk=p.pk)
+
+        ai = AlgorithmImageFactory(
+            is_manifest_valid=True,
+            is_in_registry=True,
+            is_desired_version=True,
+            algorithm=alg,
+        )
+        am = AlgorithmModelFactory(algorithm=alg, is_desired_version=True)
+        AlgorithmJobFactory(algorithm_image=ai, status=4)
```

Comment:
It is better not to use the choice values as 4 is not informative to a future reader.